content/en/docs/tasks/traffic-management/egress/wildcard-egress-hosts/index.md

```diff
@@ -298,61 +317,66 @@ The SNI proxy will forward the traffic to port `443`.
     $ kubectl create configmap egress-sni-proxy-configmap -n istio-system --from-file=nginx.conf=./sni-proxy.conf
     {{< /text >}}
 
-1.  The following command will generate `istio-egressgateway-with-sni-proxy.yaml` which you can optionally edit and then deploy.
+1.  Create an `IstioOperator` CR to add a new egress gateway with SNI proxy:
 
     {{< text bash >}}
-    $ cat <<EOF | istioctl manifest generate --set values.global.istioNamespace=istio-system -f - > ./istio-egressgateway-with-sni-proxy.yaml
-    gateways:
-      enabled: true
-      istio-ingressgateway:
-        enabled: false
-      istio-egressgateway:
-        enabled: false
-      istio-egressgateway-with-sni-proxy:
-        enabled: true
-        labels:
-          app: istio-egressgateway-with-sni-proxy
-          istio: egressgateway-with-sni-proxy
-        replicaCount: 1
-        autoscaleMin: 1
-        autoscaleMax: 5
-        cpu:
-          targetAverageUtilization: 80
-        serviceAnnotations: {}
-        type: ClusterIP
-        ports:
-          - port: 443
-            name: https
-        secretVolumes:
-          - name: egressgateway-certs
-            secretName: istio-egressgateway-certs
-            mountPath: /etc/istio/egressgateway-certs
-          - name: egressgateway-ca-certs
-            secretName: istio-egressgateway-ca-certs
-            mountPath: /etc/istio/egressgateway-ca-certs
-        configVolumes:
-          - name: sni-proxy-config
-            configMapName: egress-sni-proxy-configmap
-        additionalContainers:
-        - name: sni-proxy
-          image: nginx
-          volumeMounts:
-          - name: sni-proxy-config
-            mountPath: /etc/nginx
-            readOnly: true
+    $ cat > ./egressgateway-with-sni-proxy.yaml <<EOF
+    apiVersion: install.istio.io/v1alpha1
+    kind: IstioOperator
+    spec:
+      profile: demo
+      meshConfig:
+        outboundTrafficPolicy:
+          mode: REGISTRY_ONLY
+      components:
+        egressGateways:
+        - name: istio-egressgateway-with-sni-proxy
+          enabled: true
+          label:
+            app: istio-egressgateway-with-sni-proxy
+            istio: egressgateway-with-sni-proxy
+          k8s:
+            service:
+              ports:
+              - port: 443
+                name: https
     EOF
     {{< /text >}}
 
-1.  Deploy the new egress gateway:
+1.  Deploy the new gateway:
 
     {{< text bash >}}
-    $ kubectl apply -f ./istio-egressgateway-with-sni-proxy.yaml
-    serviceaccount "istio-egressgateway-with-sni-proxy-service-account" created
-    role "istio-egressgateway-with-sni-proxy-istio-system" created
-    rolebinding "istio-egressgateway-with-sni-proxy-istio-system" created
-    service "istio-egressgateway-with-sni-proxy" created
-    deployment "istio-egressgateway-with-sni-proxy" created
-    horizontalpodautoscaler "istio-egressgateway-with-sni-proxy" created
+    $ istioctl install -f ./egressgateway-with-sni-proxy.yaml
```

Comment:
I received the following error when tried to run this command:
```
Error: failed to install manifests: generated config failed semantic validation: port https/443 in gateway istio-egressgateway-with-sni-proxy invalid: targetPort is set to 0, which requires root. Set targetPort to be greater than 1024 or configure values.gateways.istio-egressgateway.runAsRoot=true
```

Running
 ```
istioctl install -f ./egressgateway-with-sni-proxy.yaml --set values.gateways.istio-egressgateway.runAsRoot=true
``` 
worked.